src/common/snippets/src/pass/propagate_precision.cpp

```diff
@@ -30,19 +30,12 @@ bool ov::snippets::pass::PropagatePrecision::run_on_model(const std::shared_ptr<
     bool was_updated = false;
     for (const auto& op : f->get_ordered_ops()) {
         auto type_info = op->get_type_info();
-        std::set<ov::element::TypeVector> supported_precisions;
-        // TODO: At the moment Softmax is decomposed on Linear IR level.
-        //       When Softmax will be decomposed on openvino level, remove it
-        if (type_info.is_castable(ov::op::v1::Softmax::get_type_info_static())) {
-            supported_precisions = {{ov::element::f32}};
-        } else {
-            OPENVINO_ASSERT(
-                target_machine->has(type_info),
-                "operation '" + std::string(type_info.version_id) + "::" + std::string(type_info.name) + "' was not found in target machine");
+        OPENVINO_ASSERT(
+            target_machine->has(type_info),
+            "operation '" + std::string(type_info.version_id) + "::" + std::string(type_info.name) + "' was not found in target machine");
 
-            auto exec = target_machine->get_supported_precisions(type_info);
-            supported_precisions = exec(op);
-        }
+        auto exec = target_machine->get_supported_precisions(type_info);
```

Comment:
Same question: `get_supported_precisions` will throw if `target_machine->has(...) == false`, so why do we need the exception above (on L33-25)?

BTW I think it's worth to update the exception message inside `get_supported_precisions(...)`, so it will be different from the one in `get(...)`. This way we'll know where exactly the exception was produced before even looking on the call stack, since both `get_supported_precisions(...)` and `get(...)` are called from just a couple places.

Reply:
Agree with you: the check on L33-35 is useless. I removed it and updated the exception message in `get_supported_precisions(...)`